egs/wsj/s5/steps/nnet3/tdnn/train_raw_nnet.sh

```diff
@@ -0,0 +1,626 @@
+      $cmd $dir/log/average.$x.log \
+        nnet3-average $nnets_list $dir/$[x+1].raw || exit 1;
+    else
+      # choose the best from the different jobs.
+      n=$(perl -e '($nj,$pat)=@ARGV; $best_n=1; $best_logprob=-1.0e+10; for ($n=1;$n<=$nj;$n++) {
+          $fn = sprintf($pat,$n); open(F, "<$fn") || die "Error opening log file $fn";
+          undef $logprob; while (<F>) { if (m/log-prob-per-frame=(\S+)/) { $logprob=$1; } }
+          close(F); if (defined $logprob && $logprob > $best_logprob) { $best_logprob=$logprob;
+          $best_n=$n; } } print "$best_n\n"; ' $this_num_jobs $dir/log/train.$x.%d.log) || exit 1;
+      [ -z "$n" ] && echo "Error getting best model" && exit 1;
+      $cmd $dir/log/select.$x.log \
+        nnet3-copy $dir/$[$x+1].$n.raw $dir/$[$x+1].raw || exit 1;
+    fi
+
+    rm $nnets_list
+    [ ! -f $dir/$[$x+1].raw ] && exit 1;
+    if [ -f $dir/$[$x-1].raw ] && $cleanup && \
+       [ $[($x-1)%100] -ne 0  ] && [ $[$x-1] -lt $first_model_combine ]; then
+      rm $dir/$[$x-1].raw
+    fi
+  fi
+  x=$[$x+1]
+  num_archives_processed=$[$num_archives_processed+$this_num_jobs]
+done
+
+if [ $stage -le $num_iters ]; then
+  echo "Doing final combination to produce final.raw"
+
+  # Now do combination.  In the nnet3 setup, the logic
+  # for doing averaging of subsets of the models in the case where
+  # there are too many models to reliably esetimate interpolation
+  # factors (max_models_combine) is moved into the nnet3-combine
+  nnets_list=()
+  for n in $(seq 0 $[num_iters_combine-1]); do
+    iter=$[$first_model_combine+$n]
+    nnet=$dir/$iter.raw
+    [ ! -f $nnet ] && echo "Expected $nnet to exist" && exit 1;
+    nnets_list[$n]=$nnet
+  done
+
+  # Below, we use --use-gpu=no to disable nnet3-combine-fast from using a GPU,
+  # as if there are many models it can give out-of-memory error; and we set
+  # num-threads to 8 to speed it up (this isn't ideal...)
+
+  $cmd $combine_queue_opt $dir/log/combine.log \
+    nnet3-combine --num-iters=40 \
+    --enforce-sum-to-one=true --enforce-positive-weights=true \
+    --verbose=3 "${nnets_list[@]}" "ark:nnet3-merge-egs --minibatch-size=1024 ark:$egs_dir/combine.egs ark:-|" \
+    $dir/final.raw || exit 1;
+
+  # Compute the probability of the final, combined model with
+  # the same subset we used for the previous compute_probs, as the
+  # different subsets will lead to different probs.
+  $cmd $dir/log/compute_prob_valid.final.log \
+    nnet3-compute-prob --compute-accuracy=$compute_accuracy $dir/final.raw \
+    "ark:nnet3-merge-egs ark:$egs_dir/valid_diagnostic.egs ark:- |" &
+  $cmd $dir/log/compute_prob_train.final.log \
+    nnet3-compute-prob --compute-accuracy=$compute_accuracy $dir/final.raw \
+    "ark:nnet3-merge-egs ark:$egs_dir/train_diagnostic.egs ark:- |" &
+fi
+
+if $include_log_softmax && [ $stage -le $[$num_iters+1] ]; then
+  echo "Getting average posterior for purposes of adjusting the priors."
```

Comment:
this whole block is pointless, I think, since we don't need the priors.  Unless you need the average posterior for some other purpose.